### PR TITLE
Dataplane parser refactoring

### PR DESCRIPTION
**Pull request by tomas321**
Introduce feed field count checking before list elements access.

We have experienced IndexErrors with this parser, but couldn't locate the source of the problem regarding feed formats. These changes should be sufficient enough for tracing back the source and deals with IndexErrors.

### CODE REVIEW

File: intelmq/bots/parsers/dataplane/parser.py [hunk 1]
```diff
@@ -50,23 +50,29 @@ class DataplaneParserBot(ParserBot):
     }
 
     def parse_line(self, line, report):
+        file_format = [
```
Comment (Teebor-Choka) [inline intelmq/bots/parsers/dataplane/parser.py]: It occurs to me now that this variable could be extracted to `__init__`  in order to save the variable allocations.

File: intelmq/bots/parsers/dataplane/parser.py [hunk 2]
```diff
@@ -50,23 +50,29 @@ class DataplaneParserBot(ParserBot):
...
+                raise ValueError('Incorrect format for feed {}, found line: "{}"', event.get('feed.url'), line)
+
+            if line_contents[-1].strip() in self.CATEGORY:
+                event.update(self.CATEGORY[line_contents[-1].strip()])
             else:
-                raise ValueError('Unknown data feed %r.' % value[4].strip())
+                raise ValueError('Unknown data feed {}.'.format(line_contents[-1].strip()))
+
+            for field, setter in zip(line_contents, file_format):
+                event.add(setter[0], setter[1](field.strip()))
```
Comment (Teebor-Choka) [inline intelmq/bots/parsers/dataplane/parser.py]: What happens, when `setter[1](field.strip())` returns `None`? Should it be taken care of or does `Event.add` handle that?

**Comment by Teebor-Choka** [comment]
@wagner-certat, should you wish to add some more tests: the error messages in logs were generated by lines like these (example from the modified new bot on 2019-04-01: `'Incorrect format for feed http://dataplane.org/sshclient.txt, found line: "14080        |  Telmex Colombia S.A., CO        |"`

**Comment by wagner-certat** [comment]
Thanks, I added a test case for the broken line.

bddfaed1d

**Pull request merged**